util/grpc/dialer.go

```diff
@@ -20,24 +21,27 @@ import (
 
 func WithCustomDialer() grpc.DialOption {
 	return grpc.WithContextDialer(func(ctx context.Context, addr string) (net.Conn, error) {
+		log.Infof("Custom dialer called for address: %s", addr)
 		if runtime.GOOS == "linux" {
 			currentUser, err := user.Current()
 			if err != nil {
-				log.Fatalf("failed to get current user: %v", err)
+				return nil, fmt.Errorf("failed to get current user: %w", err)
```

Comment:
We should probably return a permanent error here to avoid retries.

```suggestion
				return nil, status.Errorf(codes.FailedPrecondition, "get current user: %v", err)
```

Needs

```
	"google.golang.org/grpc/codes"
	"google.golang.org/grpc/status"
```